x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -86,33 +90,74 @@ export const signalRulesAlertType = ({
         ruleStatusSavedObjects,
         name,
       });
-      // set searchAfter page size to be the lesser of default page size or maxSignals.
-      const searchAfterSize =
-        DEFAULT_SEARCH_AFTER_PAGE_SIZE <= params.maxSignals
-          ? DEFAULT_SEARCH_AFTER_PAGE_SIZE
-          : params.maxSignals;
+
+      const searchAfterSize = Math.min(params.maxSignals, DEFAULT_SEARCH_AFTER_PAGE_SIZE);
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      let bulkIndexResult: any;
+
       try {
-        const inputIndex = await getInputIndex(services, version, index);
-        const esFilter = await getFilter({
-          type,
-          filters,
-          language,
-          query,
-          savedId,
-          services,
-          index: inputIndex,
-        });
+        if (type === 'machine_learning') {
+          if (mlJobId == null || anomalyThreshold == null) {
+            throw new Error('Attempted to execute ML Rule, but missing jobId or anomalyThreshold');
+          }
 
-        const noReIndex = buildEventsSearchQuery({
-          index: inputIndex,
-          from,
-          to,
-          filter: esFilter,
-          size: searchAfterSize,
-          searchAfterSortId: undefined,
-        });
+          const anomalyResults = await findMlSignals(
+            mlJobId,
+            anomalyThreshold,
+            from,
+            to,
+            services.callCluster
+          );
+
+          const anomalyCount = anomalyResults.hits.hits.length;
+          if (anomalyCount) {
+            logger.info(
+              `Found ${anomalyCount} signals from ML anomalies for signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}", pushing signals to index "${outputIndex}"`
+            );
+          }
+
+          bulkIndexResult = await bulkCreateMlSignals({
+            someResult: anomalyResults,
+            ruleParams: params,
+            services,
+            logger,
+            id: alertId,
+            signalsIndex: outputIndex,
+            name,
+            createdBy,
+            createdAt,
+            updatedBy,
+            updatedAt,
+            interval,
+            enabled,
+            tags,
+          });
+        } else {
+          if (index == null) {
+            throw new Error('Attempted to execute Query Rule, but no index was specified');
```

Comment:
Won't repeat the same comments but suggest adding as much information such as name, id, rule_id, and index to the error message for trouble shooting in case we encounter this error.